refactor(layer): replace all layer.type by layer.isLayer

zarov opened the pull request:
In 36c6d03 (PR #951), the isXXXLayer has been added, but wasn't used
everywhere. The change has been reflected in this commit to remove all
occurence of layer.type, leaving only the use for the debug layer (to be
removed).

DEPRECATION NOTICE: type is no longer a property of a Layer

gchoqueux commented on test/examples/.eslintrc.js:
```diff
@@ -54,7 +54,8 @@ module.exports = {
-        
+
         'prefer-destructuring': 'off',
+        'max-len': 'off',
```
could you move it in new commit?

zarov replied:
Done !

gchoqueux reviewed (approved):
thanks

zarov committed:
refactor(layer): replace all layer.type by layer.isLayer

In 36c6d03 (PR #951), the isXXXLayer has been added, but wasn't used
everywhere. The change has been reflected in this commit to remove all
occurence of layer.type, leaving only the use for the debug layer (to be
removed).

DEPRECATION NOTICE: type is no longer a property of a Layer

zarov committed:
chore(lint): remove max-length parameter for testing examples
